Diary/Model/DiaryEntityManager.swift

```diff
@@ -0,0 +1,119 @@
+                print("컨테이너 생성에 실패하였습니다.")
+            }
+        }
+        return container
+    }()
+    
+    private var context: NSManagedObjectContext {
+        return persistentContainer.viewContext
+    }
+    
+    private func saveContext() throws {
+        if context.hasChanges {
+            do {
+                try self.context.save()
+            } catch {
+                throw CoreDataError.saveContextError
+            }
+        }
+    }
+    
+    func create(diary: DiaryModel) {
+        let data = DiaryEntity(context: context)
+        
+        data.title = diary.title
+        data.body = diary.body
+        data.createdAt = diary.createdAt
+        data.id = diary.id
+
+        do {
+            try saveContext()
+        } catch CoreDataError.saveContextError {
+            print(CoreDataError.saveContextError.errorDescription)
+        } catch {
+            print("invalid error")
+        }
+    }
+    
+    func fetch() throws -> [DiaryModel] {
+        var diaryList = [DiaryEntity]()
+        do {
+            let request = DiaryEntity.fetchRequest()
+            diaryList = try context.fetch(request)
+        } catch {
+            throw CoreDataError.fetchError
+        }
+        diaryList = diaryList.reversed()
+        return diaryList.map { diaryEntity in
+            DiaryModel(
+                title: diaryEntity.title,
+                body: diaryEntity.body,
+                createdAt: diaryEntity.createdAt,
+                id: diaryEntity.id
+            )
+        }
+    }
+    
+    func update(diary: DiaryModel) {
+        let request = DiaryEntity.fetchRequest()
+        let predicate = NSPredicate(format: "id == %@", diary.id)
+
+        request.predicate = predicate
+        guard let diaryList = try? context.fetch(request) else {
+            return
+        }
+        
+        if let diaryEntity = diaryList.first {
+            diaryEntity.title = diary.title
+            diaryEntity.body = diary.body
+            diaryEntity.createdAt = diary.createdAt
+        }
+        
+        do {
+            try saveContext()
+        } catch CoreDataError.saveContextError {
+            print(CoreDataError.saveContextError.errorDescription)
+        } catch {
+            print("invalid error")
+        }
+    }
+    
+    func delete(diary: DiaryModel) throws {
+        let request = DiaryEntity.fetchRequest()
+        let predicate = NSPredicate(format: "id == %@", diary.id)
+        request.predicate = predicate
+        
+        do {
+            if let diaryEntity = try context.fetch(request).first {
+                context.delete(diaryEntity)
+            }
+        }  catch {
+            throw CoreDataError.deleteError
+        }
+
+        do {
+            try saveContext()
+        } catch CoreDataError.saveContextError {
+            print(CoreDataError.saveContextError.errorDescription)
+        } catch {
+            print("invalid error")
+        }
```

Comment:
요긴 에러를 넘길 수 있도록 `throws` 메서드로 구현되어있으니, 
```suggestion
        try saveContext()
```
만 남겨줘도 될 것 같네요.
https://docs.swift.org/swift-book/LanguageGuide/ErrorHandling.html 도 읽어보시면 좋을 것 같습니다.